dev/TeachingTip/TeachingTip.xaml

```diff
@@ -562,10 +562,10 @@
                                                     <Border x:Name="IconBorder" Grid.Column="0" Child="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.IconElement}"/>
                                                     <StackPanel x:Name="TitlesStackPanel" Grid.Column="1">
                                                         <TextBlock x:Name="TitleTextBlock" Grid.Column="0" Text="{TemplateBinding Title}" TextWrapping="WrapWholeWords"
-                                                            Foreground="{TemplateBinding Foreground}" FontFamily="{TemplateBinding FontFamily}"
+                                                            Foreground="{ThemeResource TeachingTipTitleForegroundBrush}" FontFamily="{TemplateBinding FontFamily}"
```

Comment:
Hmm, I guess I was expecting Foreground to effect the title. with your current implementation, it is only passed along to the main content's contentpresenter correct? maybe that makes more sense than title..

Reply:
No, I think its correct to have foreground affect one thing. thinking more about it I think you've done the right thing here[.

Reply:
What do you think about affecting the hero content?

Reply:
I don't think so, Hero content already has the ability to be customized (its just an object).  I don't think this would enable any more customization points.

Reply:
Technically Content is also just a property the developer has full control of, after all the developer sets that value. But I agree with your assessment here. So this PR is fine then API wise?

Reply:
and actually the same is true for the Content area of the teaching tip, so you can already acheive the above screen shot with something like:

```
<TeachingTip>
    <Grid Foreground="Aqua"/> 
</TeachingTip>
```

Thus I think I've changed my mind again, I think foreground should change the title foreground and the content foreground should be uneffected.

Reply:
But what is the point of property forwarding then? After all the main purpose of forwarding right now is that you set foreground, fontweight etc on your TeachingTip and it affects the properties.

Maybe, it is better to have it affect the Title, Content and HeroContent?

Reply:
Can you verify that if we foreward tot he content and then the consumer sets the foreground of the content the consumers value is applied?

Reply:
Setting the foreground to red on the TeachingTip and the foreground of a textblock inside the content to green yields this:

![image](https://user-images.githubusercontent.com/16122379/92151509-8d4a2700-ee21-11ea-9ad3-842ecc9a8143.png)

Reply:
Hmm does this mean that the foreground applies to the action and close buttons as well?

Reply:
Yes, that is the case, but was also the case before this PR. I think we need to think which parts of the teachingtip should take part in these APIs here. I think the close button should be excluded, the content should be included. I am not entirely sure about the icon(source), title and the hero content though.

Reply:
Agree we need to come up with a plan. Currently my thinking is that it would be ideal if the user had ability to change all of these components colors separately without retemplating, Which I think means that the foreground property can only apply to one component of the teaching tip.  Right now I'm thinking that should be the content.

Maybe applying to the icon is okay though if we are able to override that. Is it possible to set the foreground of the IconSource from the consumers markup?

Reply:
I'm afraid setting the foreground property on the iconsource does not work correctly. It seems that it gets ignored when specifying a SymbolIconSource with the Foreground property set. Setting the Foreground property on the TeachingTip itsself does color the icon source correctly though.

Reply:
I was worried about that. I think then we might want a resource for the TeachingTipIconForeground as well then.

Reply:
The reason why setting the foreground did not work is not because of the teachingtip, it is because of how IconSource works. It converts to the actual icon by returning a new control. However in that step, it forgets to pass the foreground property to those new icons. So it would work if IconSource wouldn't be forgetting to pass foreground on. Should I create a PR for the IconSource issue? I don't think that we need to make a change in TeachingTip then.

Reply:
Yeah, that sounds like a separate issue (although it is pretty similar, both controls are doing something weird which bypasses the systems automatic functionality).  If can open an issue/PR that would be awesome!

Reply:
If we assume that that fix will go in as expected, it means the only thing left for this PR is to make it so the foreground does not apply to the close and action buttons by default?

Reply:
Current state of this PR:

![image](https://user-images.githubusercontent.com/16122379/92510686-8732bc80-f20c-11ea-979e-f407dfc0aaaf.png)

The foreground of the TeachingTip was set to red in this screenshot, the action button and close button both have white foreground as expected as discussed.

Reply:
> Yeah, that sounds like a separate issue (although it is pretty similar, both controls are doing something weird which bypasses the systems automatic functionality). If can open an issue/PR that would be awesome!

I'll create a PR when I have time :)

Reply:
so in the screen shot what is causing the icon to be white?

Reply:
That was a side effect of the changes I made, fixed now:

![image](https://user-images.githubusercontent.com/16122379/92514248-e0511f00-f211-11ea-97d6-3ebaa7b11ce4.png)